src/pages/course/course.js

```diff
@@ -25,6 +26,19 @@ router.get(
     res.locals.courseNumber = courseRow.number
     res.locals.courseName = courseRow.course_name
     res.locals.course_instances = result.rows
+    // showcasing using the checks
+    if (await checks.isLoggedIn(req)) {
+      res.locals.test_perms = await (async () => {
+        return Promise.all(
+          result.rows.map(r =>
+            checks.staffHasPermissionsForCourseInstance(req, r.id)
+          )
+        )
+      })()
+    } else {
+      res.locals.test_perms = Array.from('f'.repeat(result.rows.length))
```

Comment:
Why not just do `'false\n'.repeat(result.rows.length).split('\n', result.rows.length)` instead of replacing false with f?

Example
```
'false\n'.repeat(5).split('\n', 5)
(5) ["false", "false", "false", "false", "false"]
```

Reply:
Doesn't matter since we will be changing this later but just wanted to offer a quick suggestion.

Reply:
because as soon as we actually forbid routes by auth (ie, next PR) this code will be removed and never seen again :x